### PR TITLE
implemented mariadb_binlog_checkpoint_event 

### PR DESCRIPTION
**Pull request by ebang091**
## Implemented MaridbBinLogCheckpointEvent.

- To maintain consistency, Mariadb set checkpoints to facilitate recovery in the event of an XA crash.

- MariaDB can use checkpoints to facilitate recovery when an XA crash occurs.
- MariaDB generates a binlog checkpoint event when the size of the binary log file reaches a specified value or the number of files exceeds the set value. This records the current state as a checkpoint and creates a new binary log file.

## The variables for the MariaDB checkpoint event are as follows:

:ivar filename_length: int - The length of the filename.
:ivar filename: str - The name of the file saved at the checkpoint.

## Also, Added the test code:
examine the upcoming event to be expected one (MariadbBinLogCheckpointEvent).

Thank you for reading, and I sincerely feel thankful for your pyMyReplication project. :)

### CODE REVIEW

**Comment by julien-duponchelle** [comment]
Thanks that look great , we will need to solve the conflicts before merging.

**Comment by jaehyeonpy** [comment]
I've submitted a commit that fixes the merge conflicts, but I need some time to review if other participants have anything to add. I'll inform you once the review is complete in the upcoming comment.

File: pymysqlreplication/tests/base.py [hunk 1]
```diff
@@ -144,4 +144,9 @@ def setUp(self):
         self.connect_conn_control(db)
         self.stream = None
         self.resetBinLog()
-        
\ No newline at end of file
+    
+    def bin_log_basename(self):
+        cursor = self.execute('select @@log_bin_basename')
```
Comment (ebang091) [inline pymysqlreplication/tests/base.py:149]: Could you capitalize SQL Statement before merge this PR?

**Review by ebang091 — COMMENTED** [review]
lgtm

**Comment by jaehyeonpy** [comment]
I've done the job requested by @ebang901, and it does not seem to have any more requests.

@julien-duponchelle, your approval of this PR would be appreciated.

**Comment by julien-duponchelle** [comment]
Thanks I rebased your work merged it and https://github.com/julien-duponchelle/python-mysql-replication/pull/453

**Pull request closed without merging**